src/api/libcellml/generator.h

```diff
@@ -65,216 +66,22 @@ class LIBCELLML_EXPORT Generator: public Logger
-     * analyser model from the list of tracked variables.
-     *
-     * @param analyserModel The pointer to the @ref AnalyserModel which all the algebraic variables are to be untracked.
-     */
-    void untrackAllAlgebraicVariables(const AnalyserModelPtr &analyserModel);
-
-    /**
-     * @brief Track all the variables in the given @p analyserModel.
-     *
-     * Track all the variables in the given @p analyserModel. This will add all the variables in the analyser model to
-     * the list of tracked variables. This includes all the constants, computed constants, and algebraic variables.
-     *
-     * @param analyserModel The pointer to the @ref AnalyserModel which all the variables are to be tracked.
-     */
-    void trackAllVariables(const AnalyserModelPtr &analyserModel);
-
-    /**
-     * @brief Untrack all the variables in the given @p analyserModel.
-     *
-     * Untrack all the variables in the given @p analyserModel. This will remove all the variables in the analyser model
-     * from the list of tracked variables. This includes all the constants, computed constants, and algebraic variables.
-     *
-     * @param analyserModel The pointer to the @ref AnalyserModel which all the variables are to be untracked.
-     */
-    void untrackAllVariables(const AnalyserModelPtr &analyserModel);
-
-    /**
-     * @brief Get the number of tracked constants in the given @p analyserModel.
-     *
-     * Get the number of tracked constants in the given @p analyserModel.
-     *
-     * @param analyserModel The pointer to the @ref AnalyserModel for which to get the number of tracked constants.
-     *
-     * @return The number of tracked constants in the analyser model.
-     */
-    size_t trackedConstantCount(const AnalyserModelPtr &analyserModel);
-
-    /**
-     * @brief Get the number of untracked constants in the given @p analyserModel.
-     *
-     * Get the number of untracked constants in the given @p analyserModel.
-     *
-     * @param analyserModel The pointer to the @ref AnalyserModel for which to get the number of untracked constants.
-     *
-     * @return The number of untracked constants in the analyser model.
-     */
-    size_t untrackedConstantCount(const AnalyserModelPtr &analyserModel);
-
-    /**
-     * @brief Get the number of tracked computed constants in the given @p analyserModel.
-     *
-     * Get the number of tracked computed constants in the given @p analyserModel.
-     *
-     * @param analyserModel The pointer to the @ref AnalyserModel for which to get the number of tracked computed
-     * constants.
-     *
-     * @return The number of tracked computed constants in the analyser model.
+     * @return The @ref GeneratorVariableTracker used.
      */
-    size_t trackedComputedConstantCount(const AnalyserModelPtr &analyserModel);
+    GeneratorVariableTrackerPtr variableTracker();
```

Comment:
We are happy having `variableTracker` and not `generatorVariableTracker`? I see we have `profile` and not `generatorProfile`. Is this the only class that does this, if anybody knows off the top of their head.

Reply:
This method is to be removed, so not a concern anymore. (I will finish this PR later tonight. Still need to finish my dinner...)